test/widgets/test_quickexit.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) 2015 Tycho Andersen
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# Widget specific tests
+
+import pytest
+
+import libqtile.bar
+import libqtile.config
+import libqtile.confreader
+import libqtile.layout
+from libqtile import widget
+from libqtile.ipc import IPCError
+
+
+class MinimalConf(libqtile.confreader.Config):
```

Comment:
I wonder if we can just have one copy of this and import it everywhere?

Reply:
Yup. I do use it a lot and, unfortunately, it's a bit hacky as I add the screen/bar/widget in the test function.

There's probably a much neater way of doing this but I'm still very new to pytest.

Reply:
I don't even think you need to do it with pytest magic, just put it in conftest and then import it from there. I agree there's probably a way to do it using fixtures, but IMO less smoke and mirrors is better.

Reply:
Sure. Will take a look at that.

Reply:
Yup. Quick test and that will work fine. Will update them later. Need some sleep!